tests/test_blend.py

```diff
@@ -0,0 +1,260 @@
+        morphs = [integrated_circular_gaussian(sigma=sigma) for sigma in [0.8, 2.5, 1.1, 2.1, 1.5]]
+        # Make the second component a disk component
+        morphs[1] = scipy_convolve(morphs[1], model_psf, mode="same")
+
+        # Give the first two components the same center, and unique centers
+        # for the remaining sources
+        centers = [
+            (10, 12),
+            (10, 12),
+            (20, 23),
+            (20, 10),
+            (25, 20),
+        ]
+
+        # Create the simulated image and associated data products
+        test_data = ObservationData(bands, psfs, spectra, morphs, centers, model_psf)
+
+        # Create the Observation
+        variance = np.ones((3, 35, 35), dtype=float) * 1e-2
+        weights = 1 / variance
+        weights = weights / np.max(weights)
+        self.observation = Observation(
+            test_data.convolved,
+            variance,
+            weights,
+            psfs,
+            model_psf[None],
+            bands=bands,
+        )
+        self.data = test_data
+        self.spectra = spectra
+        self.centers = centers
+        self.morphs = morphs
+
+        components = []
+        for spectrum, center, morph, data_morph in zip(
+            self.spectra, self.centers, self.morphs, self.data.morphs
+        ):
+            components.append(
+                FactorizedComponent(
+                    bands=bands,
+                    spectrum=spectrum,
+                    morph=morph,
+                    bbox=data_morph.bbox,
+                    center=center,
+                )
+            )
+
+        sources = [Source(components[:2])]
+        sources += [Source([component]) for component in components[2:]]
+
+        self.blend = Blend(sources, self.observation)
+
+    def test_exact(self):
+        """Test that a blend model initialized with the exact solution
+        builds the model correctly
+        """
+        blend = self.blend
+        self.assertEqual(len(blend.components), 5)
+        self.assertEqual(len(blend.sources), 4)
+        self.assertBoxEqual(blend.bbox, Box(self.data.images.shape[1:]))
+        self.assertImageAlmostEqual(blend.get_model(), self.data.images)
+        self.assertImageAlmostEqual(blend.get_model(convolve=True), self.observation.images)
+        self.assertImageAlmostEqual(
+            self.observation.convolve(blend.get_model(), mode="real"),
+            self.observation.images,
+        )
+
+        # Test that the log likelihood is very small
+        assert_almost_equal([blend.log_likelihood], [0])
+
+        # Test that grad_log_likelihood updates the loss
+        self.assertListEqual(blend.loss, [])
+        blend._grad_log_likelihood()
+        assert_almost_equal(blend.loss, [0])
+
+        # Remove one of the sources and calculate the non-zero log_likelihood
+        del blend.sources[-1]
+        # Check that the log-likelihood is unchanged because no gradient
```

Comment:
Yes, I can see why this is confusing, but hopefully my response to your Blend.log_likelihood question above helps.

The initial model is an exact fit to the data, so the first `_grad_log_likelihood` call adds the new loglL (~0 in this case) to `Blend.loss`. Then we delete one of the sources so that the model no longer matches the data. But until we we calculate a gradient update to the model, `Blend.log_likelihood` returns the last value for logL, which is still ~0. Then once the new gradient update is calculated the logL is updated.